### PR TITLE
Widget Venue Search Improvements

### PR DESCRIPTION
**Pull request by adamjpena**
#29: Fixed bug where venue marker disappears when there's no parking results.

#30: Get venue coordinates before drawing components so that they don't have to be manually defined in parameters (prevents lengthy zoom from world view and bug where it gets stuck at the world view). Also reduce number of api calls to retrieve venue coordinates.

#31: Accept Venue Slug as a parameter

#32: Allow widget to load if coordinates-marker is erroneously defined as true without destination coordinates

Update default mapbox-style to one not tied to an access-token

Remove unused venue_id variable.

### CODE REVIEW

File: app/scripts/widget.js [hunk 1]
```diff
@@ -61,13 +61,36 @@ export default class ParkingWidget {
 
     this.id = this._genId();
 
-    this._drawComponents();
-    this._drawPoweredBy();
-    this.searchParking();
-    this.searchEvents();
-    if (this.components.includes('time-picker')) {
-      this._element.classList.add("inc-time-picker");
+    if ('destination-venue-id' in this.options || 'destination-venue-slug' in this.options) {
```
Comment (charleslu24) [inline app/scripts/widget.js]: can you use the function `this.isVenue()` here?

File: app/scripts/widget.js [hunk 2]
```diff
@@ -468,6 +466,7 @@ export default class ParkingWidget {
 
     // Add custom event handler to markers here'
     el.addEventListener('click', () => {
+      console.log(marker.properties.quote_id);
```
Comment (charleslu24) [inline app/scripts/widget.js]: remove console.log?

File: app/scripts/widget.js (hunk 1 above)
Comment (adamjpena) [inline app/scripts/widget.js]: Absolutely

File: app/scripts/widget.js [hunk 3]
```diff
@@ -61,13 +61,36 @@ export default class ParkingWidget {
...
-    this.searchEvents();
-    if (this.components.includes('time-picker')) {
-      this._element.classList.add("inc-time-picker");
+    if ('destination-venue-id' in this.options || 'destination-venue-slug' in this.options) {
+      let xhr = new XMLHttpRequest();
+      let url = "https://api.parkwhiz.com/v3_1/venues?fields=:default,coordinates&q=slug:"
+      if ('destination-venue-slug' in this.options) {
+        url += this.options['destination-venue-slug'];
+      } else {
+        url = "https://api.parkwhiz.com/v3_1/venues/" + this.options['destination-venue-id'] + "?fields=:default,coordinates";
```
Comment (adamjpena) [inline app/scripts/widget.js]: Good call

**Pull request merged**